overlord/snapstate/snapmgr.go

```diff
@@ -1609,6 +1611,64 @@ func (m *SnapManager) ensureDownloadsCleaned() error {
 	return nil
 }
 
+// TODO use elsewhere
+func isSeeded(st *state.State) (bool, error) {
+	// only run after we are seeded
+	var seeded bool
+	err := st.Get("seeded", &seeded)
+	if err != nil && !errors.Is(err, state.ErrNoState) {
+		return false, err
+	}
+	return seeded, nil
+}
+
+// snap downloads cache cleanup runs every 24h
+const storeCacheCleanPeriod = 24 * time.Hour
+
+func (m *SnapManager) ensureStoreDownloadsCacheCleaned() error {
```

Comment:
I can split it up in 2 parts, one adding this and ensuring periodic call to cleanup and another where the actual cache policy is changed